### PR TITLE
feat(MOSSMVP-317): Frontend Implementation of the initial features for layout customization

### PR DESCRIPTION
**Pull request by Enthceph**

### CODE REVIEW

File: view/desktop/src/layouts/AppLayout.tsx [hunk 1]
```diff
@@ -10,55 +10,306 @@ import { Resizable, ResizablePanel } from "../components/Resizable";
...
+
+  const secondarySideBarVisibility = useLayoutStore((state) => state.secondarySideBar.visibility);
+  const secondarySideBarSetWidth = useLayoutStore((state) => state.secondarySideBar.setWidth);
+  const secondarySideBarGetWidth = useLayoutStore((state) => state.secondarySideBar.getWidth);
+
+  const bottomPaneVisibility = useLayoutStore((state) => state.bottomPane.visibility);
+  const bottomPaneSetHeight = useLayoutStore((state) => state.bottomPane.setHeight);
+  const bottomPaneGetHeight = useLayoutStore((state) => state.bottomPane.getHeight);
+
+  return (
```
Comment (g10z3r) [inline view/desktop/src/layouts/AppLayout.tsx:103]: It looks like this section of code is repeated more than twice. Can we consider some form of abstraction to adhere to the DRY principle?

File: view/desktop/src/parts/HeadBar/ActionsBar.tsx [hunk 2]
```diff
@@ -6,13 +6,12 @@ import { ActionsGroup } from "@/components/ActionsGroup";
 import { invokeTauriIpc } from "@/lib/backend/tauri";
 import { useLayoutStore } from "@/store/layout";
 import { MenuItem } from "@repo/moss-desktop";
-import { cn } from "@repo/moss-ui";
+import { cn, DropdownMenu as DM } from "@repo/moss-ui";
```
Comment (g10z3r) [inline view/desktop/src/parts/HeadBar/ActionsBar.tsx:9]: Is there a critical need to use an alias here? Aliases are often a result of poor entity naming conventions

File: view/desktop/src/store/layout/index.ts [hunk 3]
```diff
@@ -1,21 +1,34 @@
 import { create } from "zustand";
 
 export interface LayoutState {
+  alignment: "center" | "justify" | "left" | "right";
```
Comment (g10z3r) [inline view/desktop/src/store/layout/index.ts:4]: Extract this into a separate type, which will be provided by the backend crate moss-uikit. I might be wrong, but it’s possible that this type already exists there. The reason is that the layout position will be stored on the backend, and we need to use a common type to synchronize changes

File: view/desktop/src/store/layout/index.ts [hunk 4]
```diff
@@ -34,10 +47,35 @@ export const useLayoutStore = create<LayoutState>()((set) => ({
           visibility,
         },
       })),
+    getWidth: () => {
+      return get().primarySideBar.width;
+    },
+  },
+  secondarySideBar: {
+    width: 255,
```
Comment (g10z3r) [inline view/desktop/src/store/layout/index.ts:55]: Is 255 the default value? If so, extract it into a global constant.

File: view/desktop/src/layouts/AppLayout.tsx [hunk 5]
```diff
@@ -10,55 +10,306 @@ import { Resizable, ResizablePanel } from "../components/Resizable";
...
+            <BottomPaneContent />
           </ResizablePanel>
         </Resizable>
       </ResizablePanel>
     </Resizable>
   );
 };
+
+//----------------------------------------------------------------------------------------------
+
```
Comment (g10z3r) [inline view/desktop/src/layouts/AppLayout.tsx:286]: Why is this needed? One file - one component

File: packages/moss-ui/src/Menu/ContextMenu/ContextMenu.tsx [hunk 6]
```diff
@@ -232,6 +234,30 @@ const ContextMenuCheckboxItem = forwardRef<CustomPrimitive.CheckboxItemElement,
   }
 );
 
+/* -------------------------------------------------------------------------------------------------
+ * ContextMenuRadioGroup
+ * -----------------------------------------------------------------------------------------------*/
```
Comment (g10z3r) [inline packages/moss-ui/src/Menu/ContextMenu/ContextMenu.tsx:239]: What is this?

File: view/desktop/src/store/layout/index.ts (hunk 3 above)
Comment (Enthceph) [reply view/desktop/src/store/layout/index.ts:4]: this?
изображение

File: view/desktop/src/store/layout/index.ts (hunk 4 above)
Comment (Enthceph) [reply view/desktop/src/store/layout/index.ts:55]: Kind of. More like it is an initial value. With this set to undefined application will look like this on init. And a number 255 is me just setting number that looks ok.
изображение

File: packages/moss-ui/src/Menu/ContextMenu/ContextMenu.tsx (hunk 6 above)
Comment (Enthceph) [reply packages/moss-ui/src/Menu/ContextMenu/ContextMenu.tsx:239]: This is copy pasted from https://github.com/radix-ui/primitives/blob/main/packages/react/context-menu/src/ContextMenu.tsx since we are basically redoing their's components with few extra steps

File: view/desktop/src/layouts/AppLayout.tsx (hunk 1 above)
Comment (Enthceph) [reply view/desktop/src/layouts/AppLayout.tsx]: Changed it in the new commit

File: view/desktop/src/store/layout/index.ts (hunk 3 above)
Comment (g10z3r) [reply view/desktop/src/store/layout/index.ts:4]: Hmm, I think no... For now, just create the type you need on the frontend, and a bit later, I’ll create an exportable type on the backend that will suit this situation

**Review by g10z3r — APPROVED** [review]

File: view/desktop/src/layouts/AppLayout.tsx (hunk 1 above)
Comment (g10z3r) [reply view/desktop/src/layouts/AppLayout.tsx]: LGTM
